lib/core.js

```diff
@@ -1745,6 +1745,11 @@ type IteratorResult<+Yield,+Return> =
 interface $Iterator<+Yield,+Return,-Next> extends $Iterable<Yield,Return,Next> {
     @@iterator(): $Iterator<Yield,Return,Next>;
     next(value?: Next): IteratorResult<Yield,Return>;
+    /**
+     * Calls a defined callback function on each element returned from this iterator.
+     * @param callbackfn A function that accepts up to three arguments. The map method calls the callbackfn function one time for each element in the array.
+     */
+    map<U>(callbackfn: (value: Yield, index: number) => U): $Iterator<U, void, Next>;
```

Comment:
Note that [TypeScript's type](https://github.com/microsoft/TypeScript/blob/d85767abfd83880cea17cea70f9913e9c4496dcc/src/lib/esnext.iterator.d.ts#L32) looks slightly wrong to me. It widens `Next` to `mixed`.